packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { styled } from '@mui/material/styles';
+import { shouldForwardProp } from '@mui/system';
+import { TreeItem2DragAndDropOverlayProps } from './TreeItem2DragAndDropOverlay.types';
+import { TreeViewItemsReorderingAction } from '../models';
+
+const TreeItem2DragAndDropOverlayRoot = styled('div', {
+  name: 'MuiTreeItem2DragAndDropOverlay',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ action?: TreeViewItemsReorderingAction | null }>({
+  position: 'absolute',
+  left: 'calc(0px - var(--TreeView-itemDepth) * var(--TreeView-indentMultiplier))',
+  display: 'flex',
+  top: 0,
+  bottom: 0,
+  right: 0,
+});
+
+const TreeItem2DragAndDropOverlayContent = styled('div', {
+  name: 'MuiTreeItem2DragAndDropOverlay',
+  slot: 'Content',
+  overridesResolver: (props, styles) => styles.content,
+  shouldForwardProp: (prop) => shouldForwardProp(prop) && prop !== 'action',
+})<{ action?: TreeViewItemsReorderingAction | null }>(({ theme }) => ({
+  flexGrow: 1,
+  variants: [
+    {
+      props: { action: 'make-child' },
+      style: {
+        marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
+        border: `2px dashed ${(theme.vars || theme).palette.primary.main}`,
+      },
+    },
+    {
+      props: { action: 'reorder-above' },
+      style: {
+        marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
+        borderTop: `2px dashed ${(theme.vars || theme).palette.primary.main}`,
+      },
+    },
+    {
+      props: { action: 'reorder-below' },
+      style: {
+        marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
+        borderBottom: `2px dashed ${(theme.vars || theme).palette.primary.main}`,
+      },
+    },
+    {
+      props: { action: 'move-to-parent' },
+      style: {
+        marginLeft:
+          'calc(var(--TreeView-indentMultiplier) * calc(var(--TreeView-itemDepth) - 1))' as any,
+        borderBottom: `2px dashed ${(theme.vars || theme).palette.primary.main}`,
+      },
+    },
```

Comment:
Leaving some suggestions here to improve the styling of the overlay. 👇 

https://github.com/mui/mui-x/assets/72460825/1e27c539-3939-4ddc-8364-6dbbe71ec880


```suggestion
    {
      props: { action: 'make-child' },
      style: {
        marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
        borderRadius: theme.shape.borderRadius,
        backgroundColor: alpha((theme.vars || theme).palette.primary.dark, 0.15),
      },
    },
    {
      props: { action: 'reorder-above' },
      style: {
        marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
        borderTop: `1px solid ${alpha((theme.vars || theme).palette.grey[900], 0.6)}`,
        ...(theme.palette.mode === 'dark' && {
          borderTop: `1px solid ${alpha((theme.vars || theme).palette.grey[100], 0.6)}`,
        }),
      },
    },
    {
      props: { action: 'reorder-below' },
      style: {
        marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
        borderBottom: `1px solid ${alpha((theme.vars || theme).palette.grey[900], 0.6)}`,
        ...(theme.palette.mode === 'dark' && {
          borderBottom: `1px solid ${alpha((theme.vars || theme).palette.grey[100], 0.6)}`,
        }),
      },
    },
    {
      props: { action: 'move-to-parent' },
      style: {
        marginLeft:
          'calc(var(--TreeView-indentMultiplier) * calc(var(--TreeView-itemDepth) - 1))' as any,
        borderBottom: `1px solid ${alpha((theme.vars || theme).palette.grey[900], 0.6)}`,
        ...(theme.palette.mode === 'dark' && {
          borderBottom: `1px solid ${alpha((theme.vars || theme).palette.grey[900], 0.6)}`,
        }),
      },
    },
```

Reply:
I pushed your CSS :+1: